src/Compile.hs

```diff
@@ -364,16 +563,26 @@ instance Compilable Constructor where
       in  "({ __constructor: \"" <> n <> "\", __args: [ " <> argStr <> " ] })"
 
 
-compileImport :: FilePath -> FilePath -> FilePath -> Import -> String
-compileImport rootPath outputPath astPath (NamedImport names path absPath) =
-  let importPath = buildImportPath outputPath rootPath astPath absPath
-  in  "import { " <> compileNames names <> " } from \"" <> importPath <> "\""
-  where compileNames names = (init . init . concat) $ (++ ", ") <$> names
-
-compileImport rootPath outputPath astPath (DefaultImport alias path absPath) =
-  let importPath = buildImportPath outputPath rootPath astPath absPath
-  in  "import " <> alias <> " from \"" <> importPath <> "\""
+compileImport :: CompilationConfig -> Import -> String
+compileImport config imp =
+  let rootPath   = ccrootPath config
+      outputPath = ccoutputPath config
+      astPath    = ccastPath config
+  in  case imp of
+        NamedImport names _ absPath ->
+          let importPath = buildImportPath outputPath rootPath astPath absPath
+          in  "import { "
+                <> compileNames names
+                <> " } from \""
+                <> importPath
+                <> "\""
+         where
+          compileNames names = (init . init . concat) $ (++ ", ") <$> names
+        DefaultImport alias _ absPath ->
+          let importPath = buildImportPath outputPath rootPath astPath absPath
+          in  "import " <> alias <> " from \"" <> importPath <> "\""
 
+buildImportPath :: FilePath -> FilePath -> FilePath -> FilePath -> FilePath
```

Comment:
From this signature, I get the sense that these types can be better encapsulated?

Reply:
Yeah, mainly the names given to the params in the implementation are way clearer. But that could mostly just be packed in the config. I think all of it is in the config except the absolute path.

Reply:
That's the function that says that if the file located at /here/file.js imports a file at /elsewhere/file.js, the import generated is "../elsewhere/file.js".